notebooks/117-optimize-preprocessing/117-optimize-preprocessing.ipynb

```diff
@@ -0,0 +1,498 @@
+{
```

Comment:
<p>from what this input shape come?</p><p>It also will be useful if you can demostrate this info you using image for better explanation how this was obtained (e.g. read input image and print its shape, and data type ) </p><p> </p><p>Maybe it is also be useful to specify how to fit input data if input shape can be dynamic (e.g. what if we want process several images with different shapes)</p>
<br/>

 _Reply via <a href='https://app.reviewnb.com/openvinotoolkit/openvino_notebooks/pull/811/discussion/'>ReviewNB</a>_ <div id='ReviewNBCommentContext-DoNotDelete' style='display:none' data-state='OPEN' data-cellIndex='14'/>

Reply:
_from what this input shape come?_ - what do you mean ? where I found the values ?
_It also will be useful if you can demostrate this info you using image for better explanation how this was obtained (e.g. read input image and print its shape, and data type )_ - done
_Maybe it is also be useful to specify how to fit input data if input shape can be dynamic (e.g. what if we want process several images with different shapes)_ - fitting image is described below in section **Load image and fit it to model input**, also a little bit about dynamic input shapes there is in **Preprocessing Steps**